doc/source/ray-observability/ray-distributed-debugger.rst

```diff
@@ -77,6 +77,50 @@ Start a Ray cluster
     After checking that `ssh -p 2222 ray@localhost` works, set up VS Code as described in the
     `VS Code SSH documentation <https://code.visualstudio.com/docs/remote/ssh>`_.
 
+  .. tab-item:: KubeRay (Code Server)
+
+    Follow the instructions in :doc:`the RayCluster quickstart <../cluster/kubernetes/getting-started/raycluster-quick-start>` to set up a cluster.
+    
+    A simpler approach is to run a browser-based VS Code (Code Server) as a sidecar container in the Ray head pod. This eliminates network connectivity issues by placing VS Code inside the Kubernetes cluster.
+
+    Add a sidecar container to the Ray head pod and configure a shared volume. Modify your Ray head pod template with the following additions:
+    
+    .. code-block:: yaml
+
+        # In your RayCluster YAML, under spec.headGroupSpec.template.spec
+        containers:
+          - name: ray-head
+            # ... your existing ray-head configuration ...
+            # Add this volumeMount:
+            volumeMounts:
+              - mountPath: /tmp/ray
+                name: shared-ray-volume
+          # Add this sidecar container:
+          - name: vscode-debugger
+            image: docker.io/onesizefitsquorum/code-server-with-ray-distributed-debugger:4.101.2
+            ports:
+              - containerPort: 8443
+            volumeMounts:
+              - mountPath: /tmp/ray
+                name: shared-ray-volume
+            env:
+              - name: DEFAULT_WORKSPACE
+                value: "/tmp/ray/session_latest/runtime_resources"
+        # Add this volume at the same level as `containers`:
+        volumes:
+          - name: shared-ray-volume
+            emptyDir: {}
+
+    After the Ray cluster is running, forward the Code Server port:
+
+    .. code-block:: bash
+
+        kubectl port-forward pod/<ray-head-pod-name> 8443:8443
+
+    Access VS Code in your browser at http://127.0.0.1:8443 and use the Ray Distributed Debugger extension to connect to http://127.0.0.1:8265.
+
+    For more details, see the `Code Server with Ray Distributed Debugger <https://github.com/OneSizeFitsQuorum/Code-Server-With-Ray-Distributed-Debugger/blob/main/README.en.md>`_ project.
```

Comment:
In the readme, all the image links seem to be broken, can you fix them?

<img width="1157" height="424" alt="Screenshot 2025-08-04 at 4 03 25 PM" src="https://github.com/user-attachments/assets/53b74c4d-2e3e-47bd-b0d5-62fa42e28a73" />

Reply:
<img width="1445" height="1212" alt="image" src="https://github.com/user-attachments/assets/1f222c5a-ee79-4ad1-bca4-33ac6a82374c" />
I can access them all, they are all uploaded to github image bed in [issue](https://github.com/OneSizeFitsQuorum/Code-Server-With-Ray-Distributed-Debugger/issues/1), could you help me to check the url in your environment?

Reply:
@OneSizeFitsQuorum I also cannot see those images

Reply:
@jjyao fixed~ Please take a look again!

Reply:
Thanks, it's fixed 🎉